src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1604,33 +1162,26 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
         const auto ptr = args.Pointer();
         const auto point = args.GetCurrentPoint(*this);
+        const auto cursorPosition = point.Position();
+        const auto terminalPosition = _GetTerminalPosition(cursorPosition);
+        const auto type = ptr.PointerDeviceType();
 
         _ReleasePointerCapture(sender, args);
 
-        if (ptr.PointerDeviceType() == Windows::Devices::Input::PointerDeviceType::Mouse || ptr.PointerDeviceType() == Windows::Devices::Input::PointerDeviceType::Pen)
+        if (type == Windows::Devices::Input::PointerDeviceType::Mouse ||
+            type == Windows::Devices::Input::PointerDeviceType::Pen)
         {
-            // Short-circuit isReadOnly check to avoid warning dialog
-            if (!_isReadOnly && _CanSendVTMouseInput())
-            {
-                _TrySendMouseEvent(point);
-                args.Handled(true);
-                return;
-            }
-
-            // Only a left click release when copy on select is active should perform a copy.
-            // Right clicks and middle clicks should not need to do anything when released.
-            if (_settings.CopyOnSelect() && point.Properties().PointerUpdateKind() == Windows::UI::Input::PointerUpdateKind::LeftButtonReleased && _selectionNeedsToBeCopied)
-            {
-                CopySelectionToClipboard(false, nullptr);
-            }
+            _interactivity->PointerReleased(TermControl::GetPressedMouseButtons(point),
+                                            TermControl::PointerToMouseButtons(point),
+                                            ControlKeyStates(args.KeyModifiers()),
+                                            _focused,
+                                            terminalPosition);
         }
-        else if (ptr.PointerDeviceType() == Windows::Devices::Input::PointerDeviceType::Touch)
+        else if (type == Windows::Devices::Input::PointerDeviceType::Touch)
         {
-            _touchAnchor = std::nullopt;
+            _interactivity->TouchReleased();

```

Comment:
oh i see. then maybe `Touch` should be `TouchPressed` or something just to disambiguate.